### DISCUSSION TITLE
Possible to deploy nitro server to Vercel (without Nuxt?)

### DISCUSSION THREAD

**Discussion by outofthisworld** [nitrojs/nitro#2075] [Q&A]
### Question

I'm currently custom configuring a project that is potentially a little different from the usual as it is using two frameworks (remix.js, nitro server).

The server is packaged in the project as a seperate node_module so the server can be used across projects. Nitro (which is used for Nuxt3, has a preset called `vercel`) which produces a build output suitable for vercel.

Within the index route of the nitro server, I have created a request handler using `createRequestHandler` which is exported from "@vercel/remix". 

This route imports the remix build and does the appropriate transformations of the request to a web request for vercel.

e.g
```
export default eventHandler(async (event) => {
  // createRequestHandler imported from @vercel/remix and` importBuild()` imports the compiled remix build 
  return createRequestHandler(await importBuild())(toWebRequest(event), event.context);
});
```

The problem is that when I deploy, Vercel doesn't seem to be detecting the custom nitro server build and instead detects the remix build and runs that using Vercel. Even if I change the project type to "other" (as nitro is not available as an option) it still detects the project as a Remix project (maybe because of the lack of `nitro.config.js file` in the project root as this is within the server inside node_modules).

Any ideas on how I can get this to work? 

I was hoping to use nitro to have some of the epic benefits of using Nuxt, but using React as the front-end instead of Vue. The hard part is configuring things right so that Vercel detects the right project type and deploys things correctly

**Comment by pi0** [comment]
Hi. This should not be the case and might be a bug with vercel's edge. We have nitro test deployments (https://nitro-app.vercel.app/, https://nitro-app-edge.vercel.app/)

can you be able to make a reproduction or share your project?

**Reply by outofthisworld** [reply]
I think it has to do with how Vercel is detecting build outputs and the type of project. It seems to me that Vercel is trying to firstly identify the project type and prioritises it as a remix project over a Nitro server. It then identifies the server handler in the build output produced by Remix, and uses that with its own server rather than identifying that nitro build output exists and to use that as the build output.

I'll try and put a reproduction together through today and get back to you @pi0 

In the meantime if you are able to reproduce having a nitro server in node_modules powering a remix project would love to see how it is possible to get it working

**Reply by outofthisworld** [reply]
@pi0 

I have made a repo public for the remix component that I am using to test this:

https://github.com/outofthisworld/remixnitro

the nitro server is packaged on NPM as remix-nitro

Let me know if I can help in any other way

**Reply by Ext-NT** [reply]
You might try to check out the above answer, looks the best from here

**Reply by outofthisworld** [reply]
Sorry what answer are you referring too @Ext-NT ? The above comment is mine showcasing an example project where deploying it to Vercel does not work correctly.

**Reply by pi0** [reply]
(i guess it is an Ai answer 😓) thanks for sharing info. I have added to my todo list to check but probably we might need vercel support in the end. Have you tried disabling automatic framework preset in vercel dashboard btw?

**Reply by outofthisworld** [reply]
@pi0 Yes I have tried. I initially tried `Remix preset` in Vercel, which deployed the remix build output using Vercels server. Not being what I wanted, I changed the preset to `Other`. This also didn't correctly identify what the correct build output was to use. I also can't really specify the build output folder to Vercel, as it is expecting a 'front-end' build folder so specifying the nitro build output would just host these build files statically

**Reply by outofthisworld** [reply]
@pi0 did you manage to look into this?

**Comment by anthonyshew** [comment]
Nitro is now zero-configuration on Vercel.

Changelog: https://vercel.com/changelog/zero-configuration-support-for-nitro
